lib/src/rendering/pretty_qr_render_view.dart

```diff
@@ -0,0 +1,183 @@
+import 'package:pretty_qr_code/src/painting/pretty_qr_decoration_image.dart';
+import 'package:pretty_qr_code/src/rendering/pretty_qr_painting_context.dart';
+import 'package:pretty_qr_code/src/painting/extensions/pretty_qr_module_extensions.dart';
+
+/// {@template pretty_qr_code.PrettyQrRenderView}
+/// Paints a [PrettyQrDecoration] either before its child paints.
+/// {@endtemplate}
+@sealed
+@internal
+class PrettyQrRenderView extends RenderProxyBox {
+  /// {@template pretty_qr_code.PrettyQrRenderView.qrImage}
+  /// The QR to display.
+  /// {@endtemplate}
+  @nonVirtual
+  QrImage _qrImage;
+
+  /// {@template pretty_qr_code.PrettyQrRenderView.decoration}
+  /// What decoration to paint.
+  /// {@endtemplate}
+  @nonVirtual
+  PrettyQrDecoration _decoration;
+
+  /// {@template pretty_qr_code.PrettyQrRenderView.configuration}
+  /// The settings to pass to the decoration when painting, so that it can
+  /// resolve images appropriately. See [ImageProvider.resolve].
+  /// {@endtemplate}
+  @nonVirtual
+  ImageConfiguration _configuration;
+
+  /// The painter for a [PrettyQrDecorationImage].
+  @protected
+  DecorationImagePainter? _decorationImagePainter;
+
+  /// Creates a pretty qr view.
+  PrettyQrRenderView({
+    required QrImage qrImage,
+    required PrettyQrDecoration decoration,
+    final RenderBox? child,
+    final ImageConfiguration configuration = ImageConfiguration.empty,
+  })  : _qrImage = qrImage,
+        _decoration = decoration,
+        _configuration = configuration,
+        super(child);
+
+  /// {@macro pretty_qr_code.PrettyQrRenderView.qrImage}
+  QrImage get qrImage {
+    return _qrImage;
+  }
+
+  /// Sets the qr image.
+  set qrImage(QrImage value) {
+    if (_qrImage == value) return;
+
+    _qrImage = value;
+    markNeedsPaint();
+  }
+
+  /// {@macro pretty_qr_code.PrettyQrRenderView.decoration}
+  PrettyQrDecoration get decoration {
+    return _decoration;
+  }
+
+  /// Sets the current decoration.
+  set decoration(PrettyQrDecoration value) {
+    if (_decoration == value) return;
+
+    if (_decoration.image?.image != value.image?.image) {
+      _decorationImagePainter?.dispose();
+      _decorationImagePainter = null;
+    }
+
+    _decoration = value;
+    markNeedsPaint();
+  }
+
+  /// {@macro pretty_qr_code.PrettyQrRenderView.configuration}
+  ImageConfiguration get configuration {
+    return _configuration;
+  }
+
+  /// Sets the image configuration.
+  set configuration(ImageConfiguration value) {
+    if (value == _configuration) return;
+
+    _configuration = value;
+    markNeedsPaint();
+  }
+
+  @override
+  bool hitTestSelf(Offset position) {
+    return (Offset.zero & size).contains(position);
+  }
+
+  @override
+  Size computeSizeForNoChild(BoxConstraints constraints) {
+    return Size.square(min(constraints.maxWidth, constraints.maxHeight));
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    final canvas = context.canvas;
+    canvas.save();
```

Comment:
Do we need here to call setIsComplexHint ?

Reply:
I think it would be a good idea to consider moving this out to `PrettyQrPaintingContext`, like the original `PaintingContext` in Flutter. However, I'm not entirely convinced of its necessity right now.